clients/js/src/embeddings/transformers/TransformersEmbeddingFunction.ts

```
@@ -0,0 +1,98 @@
+    super(options, { TransformersApi: transformersApi });
+
+    if(transformersApi && !transformersApi.pipeline && this.modules){
+      // We assume, that an initialized transformersApi.pipeline has been passed instead of the module.
+      this.modules.pipeline = transformersApi.pipeline;
+    }
+  }
+
+  public async init(): Promise<void> {
+    // Define package name here because import('@xenova/transformers') throws typescript compilation errors because @xenova/transformers@2.5.3 exports broken type definitions.
+    const packageName = '@xenova/transformers';
+    try {
+      this.modules = {
+        TransformersApi: await import(packageName),
+      }
+
+      if (!this.options) {
+        throw '[TransformersEmbeddingFunction] Initializing the TransformersApi pipline failed: this.options is undefined.';
+      }
+      
+      this.modules.pipeline = await this.modules.TransformersApi.pipeline("feature-extraction", this.options.model, {
+        quantized: this.options.quantized,
+        revision: this.options.revision,
+        progress_callback: this.options.progress_callback,
+      })
+    } catch (err) {
+      console.warn('[TransformersEmbeddingFunction] Initializing the TransformersApi pipline failed. Please install the transformers package to use the TransformersEmbeddingFunction, `npm install -S @xenova/transformers`', err)
+      throw err;
+    }
+  }
+
+  public async generate(texts: string[]): Promise<number[][]> {
+    // Initialize if user fotgot to initialize
+    if (!this.modules?.pipeline) {
+      await this.init()
+      console.warn('[TransformersEmbeddingFunction] You forgot to call TransformersEmbeddingFunction#init. Will call it now to be able to generate. It is recommended to pass the transformders module via constructor.')
+    }
+
+    if (!this.modules?.pipeline) {
+      console.warn('[TransformersEmbeddingFunction] Something went wrong. The TransformersApi pipline is undefined.')
```

Maybe throw error here instead of warn? Otherwise the next step will break.

---

This should hopefully never happen, of course.